Re-structure cairo writer

rodrodros opened the pull request:
Split cairoWriter script (over 1000 lines) into several scripts

rodrodros committed:
Re-structure cairo writer

rodrodros committed:
Merge branch 'develop' into restructure-cairo-writer

piwonskp commented:
Ci does not pass. Btw consider a directory structure similiar to [solc-typed-ast](https://github.com/ConsenSys/solc-typed-ast/tree/master/src/ast/implementation). Perhaps a separate directory for statements/expressions would be helpful?

rodrodros committed:
Update cairo writer for interfaces

-> stablish a single source of truth for @interface suffix
-> interfaces which are treated as main contract don't have their names
mangled

rodrodros committed:
Update drop source unit conditions

rodrodros commented:
> Btw consider a directory structure similiar to [solc-typed-ast](https://github.com/ConsenSys/solc-typed-ast/tree/master/src/ast/implementation). Perhaps a separate directory for statements/expressions would be helpful?

I would generally agree, but in this case I think it's not that beneficial because writers are not that different at implementation level.

They all define a `writeInner` and a `writeOuter` methods and they don't receive  different treatments depending if they are declarations, statements or expressions.

rodrodros committed:
Linting

cicr99 commented on src/cairoWriter/index.ts:
```diff
@@ -0,0 +1,155 @@
+import { AST } from '../export';
```
make sure to always import from the source file instead of the export file.

cicr99 replied:
This happens on other files as well

cicr99 commented on src/cairoWriter/writers/sourceUnitWriter.ts:
```diff
@@ -0,0 +1,84 @@
+import { CairoASTNodeWriter } from '../base';
+import { writeImports } from '../utils';
+
+// Used by CairoContractWriter
```
`functionCallWriter` also uses `interfaceNameMappings`

cicr99 reviewed (commented):
Really nice work!!

rodrodros committed:
minor improvements

rodrodros committed:
imports
